packages/clerk-js/src/ui/components/UserProfile/UsernameForm.tsx

```diff
@@ -26,8 +26,7 @@ export const UsernameForm = withCardStateProvider((props: UsernameFormProps) =>
 
   const isUsernameRequired = userSettings.attributes.username.required;
 
-  const canSubmit =
-    (isUsernameRequired ? usernameField.value.length > 1 : true) && user.username !== usernameField.value;
+  const canSubmit = user.username !== usernameField.value;
```

Comment:
❓ It seems like we have the userSettings available and thus the username min and max length. Is it possible to use them on the client side validation in place of the hardcoded value?

Reply:
We have it but we rely only on the backend validation if we have username as missing field during sign-up. 
I removed the client side validation in order to have consistency and have the same behaviour in both cases

ss from missing fields screen

![Screenshot 2024-09-30 at 12 02 48 PM](https://github.com/user-attachments/assets/26076db7-b85e-4085-9cc6-1358d7162c20)

Reply:
Great thanks!

Reply:
@nikospapcom I don't see why this change improves things.

Currently we want to disable the button, and don't let you submit if you don't have a value in the field or the value that you matches your current username.

Reply:
Maybe only changing `usernameField.value.length > 0` is enough ?